src/controllers/userController.js

```diff
@@ -131,3 +139,54 @@ export const verifyUser = async (req, res) => {
     return respondWithWarning(res, statusCode.internalServerError, error.message);
   }
 };
+
+export const createProfile = async (req, res) => {
+  // de-structure id out of req.auth
+  const { id } = req.auth;
+  // Add the de-structure id to req.body with a new key of userId for ease of storage.
+  const data = { ...req.body, userId: id };
+
+  try {
+    const createdProfile = await createAProfile(data);
+    respondWithSuccess(res, statusCode.created, resMessage.profileCreated, createdProfile.dataValues);
+  } catch (error) {
+    if (error.errors[0].type === 'unique violation') {
+      respondWithWarning(res, statusCode.conflict, resMessage.profileConflict);
+    }
+    respondWithWarning(res, statusCode.internalServerError, resMessage.unableToCreateProfile);
+  }
+};
+
+export const getProfile = async (req, res) => {
+  // de-structure id out of req.auth, then rename id to userId
+  const { id: userId } = req.auth;
+
+  try {
+    const profile = await getAProfile({ userId });
+    if (!profile) {
+      respondWithSuccess(res, statusCode.success, resMessage.notAProfile);
+    }
+
+    respondWithSuccess(res, statusCode.success, resMessage.profileUpdated, profile.dataValues);
+  } catch (error) {
+    respondWithWarning(res, statusCode.internalServerError, resMessage.unableToRetrieveProfile);
+  }
+};
+
+export const updateProfile = async (req, res) => {
+  // de-structure id out of req.auth, then rename id to userId
+  const { id: userId } = req.auth;
+
+  try {
+    const profile = await getAProfile({ userId });
+    if (!profile) {
+      respondWithSuccess(res, statusCode.success, resMessage.success);
+    }
+
+    const updatedProfile = await updateAProfile({ userId }, req.body);
+
+    respondWithSuccess(res, statusCode.success, resMessage.profileUpdated, updatedProfile[1][0].dataValues);
```

Comment:
`resMessage ` creation is not necessary because they're not constants. just pass the message string directly.